File: Algorithm/Other/at coder/ActivitySelectionProblem/A39/A39.js
// 標準入力の読み取り（Node.js用）
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin", "utf8").trim().split("\n");

const N = parseInt(input[0]);
const movies = [];

// 映画の時間範囲を配列に格納
for (let i = 1; i <= N; i++) {
	const [L, R] = input[i].split(" ").map(Number);
	movies.push({ start: L, end: R });
}

// 映画を終了時間の昇順でソート
movies.sort((a, b) => a.end - b.end);

let count = 0;
let currentTime = 0;

for (const movie of movies) {
	if (movie.start >= currentTime) {
		count++;
		currentTime = movie.end;
	}
}

console.log(count);

File: Algorithm/Other/at coder/ActivitySelectionProblem/A39/A39.py
import sys

def main():
    input = sys.stdin.readline
    n = int(input())
    movies = []

    for _ in range(n):
        l, r = map(int, input().split())
        movies.append((r, l))  # 終了時間でソートしたいので (終了, 開始)

    # 終了時間で昇順ソート
    movies.sort()

    count = 0
    current_time = 0

    for r, l in movies:
        if l >= current_time:
            count += 1
            current_time = r

    print(count)

if __name__ == '__main__':
    main()
